registry/tracecat_registry/integrations/microsoft_graph.py

```diff
@@ -122,3 +121,202 @@ async def send_teams_message(
+    ],
+    top: Annotated[
+        int | None,
+        Field(
+            None,
+            description="Number of messages to return per page (default 20, max 50).",
+        ),
+    ] = None,
+    expand_replies: Annotated[
+        bool, Field(False, description="Whether to expand replies for each message.")
+    ] = False,
+) -> dict[str, str]:
+    """List messages from a Teams channel.
+
+    Note: This API requires ChannelMessage.Read.All or ChannelMessage.Read.Group permissions.
+    """
+    token = secrets.get("MICROSOFT_ACCESS_TOKEN")
+
+    headers = {"Authorization": f"Bearer {token}"}
+
+    url = f"https://graph.microsoft.com/beta/teams/{team_id}/channels/{channel_id}/messages"
+
+    params = {}
+    if top:
+        if top > 50:
+            raise ValueError("Top parameter cannot exceed 50 messages per page")
+        params["$top"] = top
+
+    if expand_replies:
+        params["$expand"] = "replies"
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, headers=headers, params=params)
+        response.raise_for_status()
+        return response.json()
+
+
+@registry.register(
+    default_title="Get user ID by email",
+    description="Get a user's ID by searching for their email address in mail or userPrincipalName fields.",
+    display_group="Microsoft Graph",
+    doc_url="https://learn.microsoft.com/en-us/graph/api/user-list?view=graph-rest-beta&tabs=http",
+    namespace="tools.microsoft_graph",
+    secrets=[microsoft_oauth_secret],
+)
+async def get_user_id_by_email(
+    email: Annotated[str, Field(..., description="The email address to search for.")],
+) -> dict[str, str]:
+    """Get a user's ID by email address.
+
+    Note: This API requires User.ReadBasic.All, User.Read.All, or Directory.Read.All permissions.
+    """
+    token = secrets.get("MICROSOFT_ACCESS_TOKEN")
+
+    headers = {"Authorization": f"Bearer {token}"}
+
+    url = "https://graph.microsoft.com/beta/users"
+
+    filter_query = f"mail eq '{email}'"
+
+    params = {"$filter": filter_query, "$select": "id", "$top": "1"}
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, headers=headers, params=params)
+        response.raise_for_status()
+        return response.json()
+
+
+@registry.register(
+    default_title="Delete Teams channel",
+    description="Delete a channel from a Microsoft Teams team.",
+    display_group="Microsoft Graph",
+    doc_url="https://learn.microsoft.com/en-us/graph/api/channel-delete?view=graph-rest-beta&tabs=http",
+    namespace="tools.microsoft_graph",
+    secrets=[microsoft_oauth_secret],
+)
+async def delete_teams_channel(
+    team_id: Annotated[
+        str, Field(..., description="The ID of the team containing the channel.")
+    ],
+    channel_id: Annotated[
+        str, Field(..., description="The ID of the channel to delete.")
+    ],
+) -> dict[str, str]:
```

Comment:
is this actually `dict[str,str]` if you're returning `response.status_code` in one of the values?